gnucash/gnome/assistant-stock-transaction.cpp

```diff
@@ -0,0 +1,1151 @@
+        break;
+    case PAGE_CASH:
+        refresh_page_cash (info->cash_value, info);
+        break;
+    case PAGE_FEES:
+        refresh_page_fees (info->fees_value, info);
+        break;
+    case PAGE_DIVIDEND:
+        refresh_page_dividend (info->fees_value, info);
+        break;
+    case PAGE_CAPGAINS:
+        refresh_page_capgains (info->capgains_value, info);
+        break;
+    case PAGE_FINISH:
+        refresh_page_finish (info);
+        break;
+    }
+}
+
+static gint
+forward_page_func (gint current_page, gpointer user_data)
+{
+    auto info = (StockTransactionInfo *)user_data;
+    auto& txn_type = info->txn_type;
+
+    current_page++;
+    if (txn_type.stock_amount == DISABLED && current_page == PAGE_STOCK_AMOUNT)
+        current_page++;
+    if (txn_type.stock_value == DISABLED && current_page == PAGE_STOCK_VALUE)
+        current_page++;
+    if (txn_type.cash_value == DISABLED && current_page == PAGE_CASH)
+        current_page++;
+    if (txn_type.fees_value == DISABLED && current_page == PAGE_FEES)
+        current_page++;
+    if (txn_type.dividend_value == DISABLED && current_page == PAGE_DIVIDEND)
+        current_page++;
+    if (txn_type.capgains_value == DISABLED && current_page == PAGE_CAPGAINS)
+        current_page++;
+
+    return current_page;
+}
+
+using AccountVec = std::vector<Account*>;
+
+static void create_split (Transaction *trans, uint splitfield,
+                          const gchar *action, Account *account,
+                          AccountVec& account_commits, GtkWidget *memo_entry,
+                          GtkWidget *amount, GtkWidget *value,
+                          bool skip_if_zero)
+{
+    auto amount_numeric = amount ? gnc_amount_edit_get_amount (GNC_AMOUNT_EDIT (amount)) : gnc_numeric_zero ();
+    auto value_numeric = value ? gnc_amount_edit_get_amount (GNC_AMOUNT_EDIT (value)) : gnc_numeric_zero ();
+
+    if (skip_if_zero && gnc_numeric_zero_p (value_numeric))
+        return;
+
+    if (splitfield & ENABLED_CREDIT)
+    {
+        amount_numeric = gnc_numeric_neg (amount_numeric);
+        value_numeric = gnc_numeric_neg (value_numeric);
+    }
+
+    auto split = xaccMallocSplit (gnc_get_current_book ());
+    xaccSplitSetParent (split, trans);
+    xaccAccountBeginEdit (account);
+    account_commits.emplace_back (account);
+    xaccSplitSetAccount (split, account);
+    xaccSplitSetMemo (split, gtk_entry_get_text (GTK_ENTRY (memo_entry)));
+    xaccSplitSetValue (split, value_numeric);
+    xaccSplitSetAmount (split, amount_numeric);
+    // g_warning ("creating %s split in account %s with Val(%s), Amt(%s). result Val(%s), Amt(%s)",
+    //            action, xaccAccountGetName (account),
+    //            gnc_numeric_to_string (value_numeric),
+    //            gnc_numeric_to_string (amount_numeric),
+    //            gnc_numeric_to_string (xaccSplitGetValue (split)),
+    //            gnc_numeric_to_string (xaccSplitGetAmount (split)));
```

Comment:
Either delete this or uncomment it and change `g_warning` to `DEBUG`. Always prefer the libgnucash/core-utils/qoflog.h macros to calling GLib logging functions directly.